File: in-class-exercises/closures/make_multiplier.py
# Use named function
def make_multiplier(by):
    def multilpy(x):
        return x * by
    return multilpy

doubler = make_multiplier(2)

print(doubler(5)) #10
print(doubler(7)) #14

# Use lambda
def make_multiplier(by):
    return lambda x: x * by

doubler = make_multiplier(2)

print(doubler(5)) #10
print(doubler(7)) #14

File: inClassExercises/closures/make_multiplier.js
// Use named function
function make_multiplier(by) {
	function multilpy(x) {
		return x * by;
	}
	return multilpy;
}

doubler = make_multiplier(2);

console.log(doubler(5)); //10
console.log(doubler(7)); //14

// Use lambda
function make_multiplier(by) {
	return (x) => x * by;
}

doubler = make_multiplier(2);

console.log(doubler(5)); //10
console.log(doubler(7)); //14
